File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.py
class Solution(object):
    def uniqueOccurrences(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        my_dict = {}
        for item in arr:
            if my_dict.get(item,-1)==-1:
                my_dict[item]=1
            else:
                my_dict[item]+=1
        
        values = list(my_dict.values())
        if len(values) != len(set(values)):
            return False
        else:
            return True

File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
       Map<Integer, Integer> countMap = new HashMap<>();
        for (int i=0;i<arr.length;i++) {
            countMap.put(arr[i], countMap.getOrDefault(arr[i], 0) + 1);
        }

        Set<Integer> occurrenceSet = new HashSet<>();
        
        for (int count : countMap.values()) {
            if (!occurrenceSet.add(count)) {
                return false;
            }
        }

        return true; 
    }
}